Chat/Program.cs

```diff
@@ -87,10 +99,41 @@ private static void ReceiveMessage()
         private static void WriteChat()
         {
             Console.Clear();
-            foreach(var message in messages)
+            foreach (var message in messages)
             {
-                Console.WriteLine(message);
+                Console.WriteLine(message.Message);
+            }
+        }
+
+        private static void RightOrder(String message)
+        {
+            int sizeOfNumber = message.IndexOf('|');
+            int index = Int32.Parse(message.Substring(0, sizeOfNumber - 1));
+            messages.OrderBy(x => x.Nomber);
+            if (messages.Count == 0)
+            {
+                messages.Add(new MessageInOrder(index, message));
+            }
+            else if (messages.LastOrDefault().Nomber == index - 1)
+            {
+                messages.Add(new MessageInOrder(index, message));
+            }
+            else
+            {
+                UdpClient sender = new UdpClient();
+                try
+                {
+                    string error = String.Format("|Your companion didn't get message no. {0} and next!", index);
```

Comment:
And now ask client to re-send lost messages to continue dialog session. Doesn't make much sense to abort it.

Reply:
Fixed. Messages wil be resend and chat will continue.